src/daemon/control.rs

```diff
@@ -1050,6 +1216,208 @@ pub fn handle_rpc_messages(
                 );
                 response_tx.send(tx_res.map_err(|e| RpcControlError::Transaction(e.into())))?;
             }
+            RpcMessageIn::UpdateSpendTx(outpoints, spend_tx, response_tx) => {
+                log::trace!("Got 'updatespendtx' request from RPC thread");
+                let revaultd = revaultd.read().unwrap();
+                let db_path = revaultd.db_file();
+                let spend_txid = spend_tx.inner_tx().global.unsigned_tx.txid();
+
+                // Fetch the Unvault it spends from the DB
+                let mut db_unvaults = Vec::with_capacity(outpoints.len());
+                for outpoint in outpoints.iter() {
+                    let db_vault = match db_vault_by_deposit(&db_path, &outpoint)? {
+                        Some(v) => v,
+                        None => {
+                            response_tx.send(Err(RpcControlError::UnknownOutpoint(*outpoint)))?;
+                            break;
+                        }
+                    };
+
+                    if !matches!(db_vault.status, VaultStatus::Active) {
+                        response_tx.send(Err(RpcControlError::InvalidStatus((
+                            db_vault.status,
+                            VaultStatus::Active,
+                        ))))?;
+                        break;
+                    }
+
+                    db_unvaults.push(
+                        db_presigned_tx(&db_path, outpoint, TransactionType::Unvault)?
+                            .expect("Vault is active, it must be fullysigned"),
+                    );
+                }
+                // There was an issue with an outpoint, error is already sent in the loop.
+                if db_unvaults.len() != outpoints.len() {
+                    continue;
+                }
+
+                // Does the Spend psbt we are passed actually correspond to the outpoint array?
+                fn all_outpoints_match(
+                    spend_inputs: &Vec<revault_tx::bitcoin::TxIn>,
```

Comment:
From `cargo clippy`:
```
warning: writing `&Vec<_>` instead of `&[_]` involves one more reference and cannot be used with non-Vec-based slices.
    --> src/daemon/control.rs:1288:35
     |
1288 |                     spend_inputs: &Vec<revault_tx::bitcoin::TxIn>,
     |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: change this to: `&[revault_tx::bitcoin::TxIn]`
     |
     = note: `#[warn(clippy::ptr_arg)]` on by default
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
```
(same for `db_unvaults`)